components/gitpod-protocol/src/util/tracing.ts

```diff
@@ -39,7 +49,42 @@ export namespace TraceContext {
             "error": err.message,
             "stacktrace": err.stack
         })
-        ctx.span.setTag("error", true)
+        ctx.span.setTag("error", true);
+    }
+
+    export function logJsonRPCError(ctx: TraceContext, method: string, err: ResponseError<any>) {
+        if (!ctx.span) {
+            return;
+        }
+        logError(ctx, err);
+
+        // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md#json-rpc
+        ctx.span.addTags({
+            rpc: {
+                system: "jsonrpc",
+                method,
+                jsonrpc: {
+                    error_code: err.code,
+                    error_message: err.message,
+                },
+            },
+        });
+    }
+
+    export function addJsonRPCParameters(ctx: TraceContext, method: string, args: any[]) {
+        if (!ctx.span) {
+            return;
+        }
+
+        ctx.span.addTags({
+            rpc: {
```

Comment:
@geropl Looks like it isn't well defined in the JS SDK for opentracing how to deal with the values ([see docs](https://opentracing-javascript.surge.sh/classes/span.html#addtags)) and it decided to deal with it poorly ([see example trace](https://ui.honeycomb.io/gitpod/datasets/gitpod-staging/result/yrWL9r8w1Cc/trace/mCLpqHV6BZ6?span=6aefff5d8ad2ce4e)) and just serialise the value for the `rpc` key as a string 😅

So either we have to change this to be multiple calls to addTag

```ts
ctx.span.addTag('rpc.system', jsonrpc')
ctx.span.addTag('rpc.method',method)
// and so on
``` 

Or create a helper function like the one below (it would need cleaning and tests of course)

```ts
function addNestedTags(tags: any, namespace?: string) {
  namespace = namespace ? `${namespace}.` : ''
  Object.entries(tags).forEach((arr) => {
    const [k,v] = arr
    // Arrays are objects, this is fine.
    if (typeof v == 'object') {
      addNestedTags(v, `${namespace}${k}`)
    }
    else {
      // delegate to span.addTag and let it deal with the rest of the cases ()
      console.log(`${namespace}${k} = ${v}`)
    }
  })
}

addNestedTags({
  rpc: {
    system: 'jsonrpc',
    method: 'foobar',
    parameters: ['test', {
      'more': 42
    }]
  }
})

// rpc.system = jsonrpc
// rpc.method = foobar
// rpc.parameters.0 = test
// rpc.parameters.1.more = 42
```

Reply:
Who on earth things this would be a good way to implement `addTags`.... :see_no_evil: 

Thx for pointing out @mads-hartmann , will fix!